### DISCUSSION TITLE
enhancement Filter patterns

### DISCUSSION THREAD

**Discussion by Pantyhose-X** [DNSCrypt/dnscrypt-proxy#1962] [Ideas]
solve complicated cases when regular ad blocking rules aren't enough

#  rules syntax `*`  `@@` `/` 
| Pattern               | Result                                                                                        |
| --------------------- | ----------------------------------------------------------------------------------------------|
|`example.org`          |block access to the example.org domain and all its subdomains                                  |
|`*.example`            |block access `.example` All prefixes  to such as `mail.example.com`                            |
|`example.*`            |block access `example.` All suffixes to such as `example.com`  `example.net`                   |
|`*sex*`                |matches any name containing that substring                                                     |
|`@@example.org`        |unblock access to the example.org domain and all its subdomains                                |
|`0.0.0.0 example.org`  |will now return 0.0.0.0 address for the example.org domain (but not its subdomains). Other IP  |
|`0.0.0.0/24`         |block access to ip range   domains  `0.0.0.0/24`                                               |
|`0.0.0.*`            |block access to  ip range Incremental domains  `0.0.0.*`                                       |
|`0.0.0.0`            |block access to all domains whose CNAME response returns this IP address                       |

<details>
<summary>details</summary>

````java
#  rules syntax `*`  `@@` `" "` 
| Pattern               | Result                                                                                        |
| --------------------- | ----------------------------------------------------------------------------------------------|
|`example.org`          |block access to the example.org domain and all its subdomains                                  |
|`*.example`            |block access `.example` All prefixes  to such as `mail.example.com`                            |
|`example.*`            |block access `example.` All suffixes to such as `example.com`  `example.net`                   |
|`*sex*`                |matches any name containing that substring                                                     |
|`@@example.org`        |unblock access to the example.org domain and all its subdomains                                |
|`0.0.0.0 example.org`  |will now return 0.0.0.0 address for the example.org domain (but not its subdomains). Other IP  |
|`0.0.0.0/24`         |block access to ip range   domains  `0.0.0.0/24`                                               |
|`0.0.0.*`            |block access to  ip range Incremental domains  `0.0.0.*`                                       |
|`0.0.0.0`            |block access to all domains whose CNAME response returns this IP address                       |
````
</details>

**Comment by jedisct1** [comment]
`dnscrypt-proxy` can [already](https://github.com/DNSCrypt/dnscrypt-proxy/wiki/Filters) block IP addresses, IPv4 and IPv6 subnets, and return configured IP addresses for specific names ([cloaking](https://github.com/DNSCrypt/dnscrypt-proxy/wiki/Cloaking)). Allow lists (to "unblock access") are also supported.

**Reply by Pantyhose-X** [reply]
> `dnscrypt-proxy` can [already](https://github.com/DNSCrypt/dnscrypt-proxy/wiki/Filters) block IP addresses, IPv4 and IPv6 subnets, and return configured IP addresses for specific names ([cloaking](https://github.com/DNSCrypt/dnscrypt-proxy/wiki/Cloaking)). Allow lists (to "unblock access") are also supported.


- Add IP range block access `/`
<html>
<body>


0.0.0.0/24 | block access to ip range domains `0.0.0.0/24`
-- | --



</body>
</html>

- I recommend using, as I mentioned, this block access prefixes suffixes

<html>
<body>


*.example | block access .example All prefixes to such as `mail.example.com`
-- | --
example.* | block access example. All suffixes to such as `example.com` `example.net`


</body>
</html>
